cli/dcoscli/cluster/main.py

```diff
@@ -0,0 +1,214 @@
+"""Provides status about the DCOS cluster managed by this CLI
+
+Usage:
+   dcos cluster --help
+   dcos cluster --info
+   dcos cluster --version
+   dcos cluster help
+   dcos cluster status [--json]
+
+Options:
+    --help                  Show this screen
+    --info                  Show info
+    --json                  Print json-formatted cluster status
+    --version               Show version
+"""
+import dcoscli
+import docopt
+from dcos import cmds, emitting, util
+from dcos.errors import DCOSException
+from dcos.http import request
+from dcoscli import tables
+
+logger = util.get_logger(__name__)
+
+emitter = emitting.FlatEmitter()
+
+MESOS_STATE_URL = '{}:5050/state.json'
+OK_STATUS = 200
+
+
+def main():
+    try:
+        return _main()
+    except DCOSException as e:
+        emitter.publish(e)
+        return 1
+
+
+def _main():
+    util.configure_logger_from_environ()
+
+    args = docopt.docopt(
+        __doc__,
+        version='dcos-status version {}'.format(dcoscli.version))
+
+    return cmds.execute(_cmds(), args)
+
+
+def _cmds():
+    """
+    :returns: All of the supported commands
+    :rtype: [Command]
+    """
+
+    return [
+        cmds.Command(
+            hierarchy=['cluster', '--info'],
+            arg_keys=[],
+            function=_info),
+
+        cmds.Command(
+            hierarchy=['cluster', 'status'],
+            arg_keys=['--json'],
+            function=_status),
+        ]
+
+
+def _status(is_json):
+    """ Print to cli cluster status
+
+    :param: args: dcos cluster input arguments
+    type: args: dict
+    """
+    cluster_data, exit_code = get_cluster_data()
+    emitting.publish_table(emitter, cluster_data['Components'],
+                           tables.cluster_table, is_json)
+    if exit_code != 0:
+        raise DCOSException()
+
+
+def _info():
+    """Print task cli information.
+
+    :returns: process return code
+    :rtype: int
+    """
+
+    emitter.publish(__doc__.split('\n')[0])
+    return 0
+
+
+def get_cluster_data():
+    """ Get information about current cli cluster
+
+    :returns: Cluster status data
+    :rtype: dict
+    """
+    exit_code_list = list()
+
+    def sum_codes(f):
+        status, exit_code = f()
+        exit_code_list.append(exit_code)
+        return status
+
+    return {'Components': [{'Name': 'Mesos Master',
```

Comment:
It looks like you only ever use the 'Components' key.  If that's the case, just remove it.  It's unnecessary.